extensions-contrib/influxdb-emitter/src/main/java/org/apache/druid/emitter/influxdb/InfluxdbEmitter.java

```diff
@@ -0,0 +1,231 @@
+
+    this.dimensionWhiteList = ImmutableSet.of(
+        "dataSource",
+        "type",
+        "numMetrics",
+        "numDimensions",
+        "threshold",
+        "dimension",
+        "taskType",
+        "taskStatus",
+        "tier"
+    );
+
+    log.info("constructing influxdb emitter");
+  }
+
+  @Override
+  public void start()
+  {
+    synchronized (started) {
+      if (!started.get()) {
+        exec.scheduleAtFixedRate(
+            new ConsumerRunnable(),
+            influxdbEmitterConfig.getFlushDelay(),
+            influxdbEmitterConfig.getFlushPeriod(),
+            TimeUnit.MILLISECONDS
+        );
+        started.set(true);
+      }
+    }
+  }
+
+  @Override
+  public void emit(Event event)
+  {
+    if (event instanceof ServiceMetricEvent) {
+      ServiceMetricEvent metricEvent = (ServiceMetricEvent) event;
+      try {
+        eventsQueue.put(metricEvent);
+      }
+      catch (InterruptedException exception) {
+        log.error(exception.toString());
+        Thread.currentThread().interrupt();
+      }
+    }
+  }
+
+  public void postToInflux(String payload)
+  {
+    HttpPost post = new HttpPost(
+        "http://" + influxdbEmitterConfig.getHostname()
+        + ":" + influxdbEmitterConfig.getPort()
+        + "/write?db=" + influxdbEmitterConfig.getDatabaseName()
+        + "&u=" + influxdbEmitterConfig.getInfluxdbUserName()
+        + "&p=" + influxdbEmitterConfig.getInfluxdbPassword()
+    );
+
+    post.setEntity(new StringEntity(payload, ContentType.DEFAULT_TEXT));
+    post.setHeader("Content-Type", "application/x-www-form-urlencoded");
+
+    try {
+      influxdbClient.execute(post);
+    }
+    catch (IOException ex) {
+      log.info(ex.toString());
+    }
+    finally {
+      post.releaseConnection();
+    }
+  }
+
+  public String transformForInfluxSystems(ServiceMetricEvent event)
+  {
+    // split Druid metric on slashes and join middle parts (if any) with "_"
+    String[] parts = getValue("metric", event).split("/");
+    String metric = String.join(
+        "_",
+        Arrays.asList(
+            Arrays.copyOfRange(
+                parts,
+                1,
+                parts.length - 1
+            )
+        )
+    );
+
+    String payload = "druid_" + parts[0] + ",";
+
+    payload += "service=" + getValue("service", event)
+               + ((parts.length == 2) ? "" : ",metric=druid_" + metric)
+               + ",hostname=" + getValue("host", event).split(":")[0];
+
+
+    ImmutableSet<String> dimNames = ImmutableSet.copyOf(event.getUserDims().keySet());
+    for (String dimName : dimNames) {
+      if (this.dimensionWhiteList.contains(dimName)) {
+        payload += "," + dimName + "=" + sanitize(String.valueOf(event.getUserDims().get(dimName)));
+      }
+    }
+
+    payload += " druid_" + parts[parts.length - 1] + "=" + getValue("value", event);
+
+    return payload + " " + event.getCreatedTime().getMillis() * 1000000 + '\n';
+  }
```

Comment:
lots of string concatenation above .. it would be better to use format or StringBuilder or something else like that.